packages/material-ui/src/Stack/Stack.js

```diff
@@ -0,0 +1,285 @@
+  'align-items-flex-end': {
+    alignItems: 'flex-end',
+  },
+  /* Styles applied to the root element if `alignItems="baseline"`. */
+  'align-items-baseline': {
+    alignItems: 'baseline',
+  },
+  /* Styles applied to the root element if `alignContent="center"`. */
+  'align-content-center': {
+    alignContent: 'center',
+  },
+  /* Styles applied to the root element if `alignContent="flex-start"`. */
+  'align-content-flex-start': {
+    alignContent: 'flex-start',
+  },
+  /* Styles applied to the root element if `alignContent="flex-end"`. */
+  'align-content-flex-end': {
+    alignContent: 'flex-end',
+  },
+  /* Styles applied to the root element if `alignContent="space-between"`. */
+  'align-content-space-between': {
+    alignContent: 'space-between',
+  },
+  /* Styles applied to the root element if `alignContent="space-around"`. */
+  'align-content-space-around': {
+    alignContent: 'space-around',
+  },
+  /* Styles applied to the root element if `justify="center"`. */
+  'justify-center': {
+    justifyContent: 'center',
+  },
+  /* Styles applied to the root element if `justify="flex-end"`. */
+  'justify-flex-end': {
+    justifyContent: 'flex-end',
+  },
+  /* Styles applied to the root element if `justify="space-between"`. */
+  'justify-space-between': {
+    justifyContent: 'space-between',
+  },
+  /* Styles applied to the root element if `justify="space-around"`. */
+  'justify-space-around': {
+    justifyContent: 'space-around',
+  },
+  /* Styles applied to the root element if `justify="space-evenly"`. */
+  'justify-space-evenly': {
+    justifyContent: 'space-evenly',
+  },
+  ...generateGutter(theme),
+});
+
+const Stack = React.forwardRef((props, ref) => {
+  const {
+    alignContent = 'stretch',
+    alignItems = 'stretch',
+    classes,
+    className: classNameProp,
+    component: Component = 'div',
+    direction = 'row',
+    item = false,
+    justify = 'flex-start',
+    spacing = 0,
+    wrap = 'nowrap',
+    zeroMinWidth = false,
+    ...other
+  } = props;
+
+  const className = clsx(
+    classes.root,
+    {
+      [classes.item]: item,
+      [classes.zeroMinWidth]: zeroMinWidth,
+      [classes[`spacing-row-${String(spacing)}`]]: spacing !== 0 && direction.indexOf('row') === 0,
+      [classes[`spacing-column-${String(spacing)}`]]: spacing !== 0 && direction.indexOf('column') === 0,
+      [classes[`direction-${String(direction)}`]]: direction !== 'row',
+      [classes[`wrap-${String(wrap)}`]]: wrap !== 'wrap',
+      [classes[`align-items-${String(alignItems)}`]]: alignItems !== 'stretch',
+      [classes[`align-content-${String(alignContent)}`]]: alignContent !== 'stretch',
+      [classes[`justify-${String(justify)}`]]: justify !== 'flex-start',
+    },
+    classNameProp,
+  );
+
+  return <Component className={className} ref={ref} {...other} />;
+});
+
+if (process.env.NODE_ENV !== 'production') {
+  // can't use named function expression since the function body references `Stack`
```

Comment:
Just curious what exactly is meant by

> the function body references `Stack` which would point to the render function instead of the actual component

because I couldn't find references to anything but `Component` in the function body?

In other words, what makes this component different from other components that are able to use `const SomeName = forwardRef(function SomeName() {props, ref})`